File: 48-rotate-image/rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        n = len(matrix)

        # transpose
        for i in range(n):
            for j in range(i, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

        #flip horizontally-clock wise rotation
        for i in range(n):
            matrix[i].reverse()

File: 48-rotate-image/rotate-image.js
var rotate = function (matrix) {
    const n = matrix.length

    // transpose
    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]]
        }
    }
    // reverse
    for (let i = 0; i < n; i++) {
        matrix[i].reverse()
    }
};